src/modules/billing/services/billing-volumes-service.js

```diff
@@ -0,0 +1,82 @@
+const billingVolumesRepo = require('../../../lib/connectors/repos/billing-volumes');
+const mappers = require('../mappers');
+const twoPartTariffMatching = require('./two-part-tariff-service');
+const { isEmpty, flatMap } = require('lodash');
+
+const chargeElementSeasonMatches = (chargeElement, isSummer) => {
+  const chargeElementIsSummer = chargeElement.season === 'summer';
+  return chargeElementIsSummer === isSummer;
+};
+
+const getChargeElementIds = (chargeVersion, isSummer) => {
+  return chargeVersion.chargeElements.filter(chargeElement => {
+    if (chargeElementSeasonMatches(chargeElement, isSummer)) {
+      return chargeElement.chargeElementId;
+    };
+  });
+};
+
+const billingVolumeToModel = volume => mappers.billingVolume.dbToModel(volume);
+
+const volumesToModels = volumes => volumes.map(billingVolumeToModel);
+
+const isMissingVolumes = volumes => {
+  const containsNull = volumes.map(volume => isEmpty(volume));
+  return containsNull.includes(true);
+};
+
+const getVolumesForChargeElements = async (chargeVersion, financialYear, isSummer) => {
+  const chargeElements = getChargeElementIds(chargeVersion, isSummer);
+  const billingVolumes = await Promise.all(
```

Comment:
We maybe want a repo method here that can accept an array of IDs (this can be done with Bookshelf collections if fetching related models) to get all the data in one go.
```
const chargeElementIds = chargeElements.map(chargeElement => chargeElement.chargeElementId);
const billingVolumes = await billingVolumesRepo.findByChargeElementIdsFinancialYearAndSeason(chargeElementIds, financialYear, isSummer);
```